transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringDatagramChannel.java

```diff
@@ -451,43 +474,109 @@ protected void readComplete0(int res) {
         }
 
         @Override
-        protected void scheduleRead0() {
+        protected int scheduleRead0() {
             final IOUringRecvByteAllocatorHandle allocHandle = recvBufAllocHandle();
             ByteBuf byteBuf = allocHandle.allocate(alloc());
-            IOUringSubmissionQueue submissionQueue = submissionQueue();
-
             assert readBuffer == null;
             readBuffer = byteBuf;
 
-            recvMsg = !isConnected();
-            long bufferAddress = byteBuf.memoryAddress();
-            allocHandle.attemptedBytesRead(byteBuf.writableBytes());
+            int writable = byteBuf.writableBytes();
+            allocHandle.attemptedBytesRead(writable);
+            int datagramSize = config().getMaxDatagramPayloadSize();
 
-            if (!recvMsg) {
-                submissionQueue.addRead(socket.intValue(), bufferAddress,
+            int numDatagram = datagramSize == 0 ? 1 : Math.max(1, byteBuf.writableBytes() / datagramSize);
```

Comment:
thats by "design" as I want to ensure the message is datagramSize... Otherwise we may receive truncated messages which may not be expected by the user.